sdk/auth/managed_identity_authorizer.go

```diff
@@ -173,9 +177,9 @@ func azureMetadata(ctx context.Context, url string) (body []byte, err error) {
 		instanceMetadataService: true,
 
 		retryWaitMin:  2 * time.Second,
-		retryWaitMax:  60 * time.Second,
-		retryMaxCount: 5,
-		timeout:       10 * time.Second,
+		retryWaitMax:  retryWaitMax * time.Second,
+		retryMaxCount: retryMaxCount,
+		timeout:       timeout * time.Second,
```

Comment:
we can use the timeout on the context:

```suggestion
		timeout:       time.Until(deadline),
```